tests/databricks_ai_bridge/test_genie.py

```diff
@@ -442,3 +454,85 @@ def test_parse_query_result_with_timestamp_formats():
+
+        mock_workspace_client.genie._api.do.side_effect = [
+            {"status": "COMPLETED", "attachments": [{"text": {"content": "Done"}}]},
+        ]
+
+        genie.poll_for_result("123", "456")
+
+        mock_start_span.assert_called_once_with(name="genie_timeline", span_type="CHAIN")
+
+
+def test_poll_for_result_span_create_on_status_change(genie, mock_workspace_client):
+    with (
+        patch("mlflow.start_span") as mock_start_span,
+        patch("mlflow.tracking.MlflowClient") as MockClient,
+        patch("time.sleep", return_value=None),
+    ):
+        mock_client = MockClient.return_value
+        mock_parent_span = MagicMock()
+        mock_parent_span.trace_id = "trace_123"
+        mock_parent_span.span_id = "parent_456"
+        mock_start_span.return_value.__enter__.return_value = mock_parent_span
+
+        mock_child_span = MagicMock()
+        mock_child_span.span_id = "child_789"
+        mock_client.start_span.return_value = mock_child_span
+
+        mock_workspace_client.genie._api.do.side_effect = [
+            {"status": "EXECUTING_QUERY"},
+            {"status": "COMPLETED", "attachments": [{"text": {"content": "Done"}}]},
+        ]
+
+        genie.poll_for_result("123", "456")
+
+        # Verify span was created for EXECUTING_QUERY
+        mock_client.start_span.assert_called_once()
+        start_kwargs = mock_client.start_span.call_args[1]
+        assert start_kwargs["name"] == "executing_query"
+        assert start_kwargs["trace_id"] == "trace_123"
+        assert start_kwargs["parent_id"] == "parent_456"
+        assert start_kwargs["span_type"] == "CHAIN"
+
+
+def test_poll_for_result_span_close_on_status_change(genie, mock_workspace_client):
+    with (
+        patch("mlflow.start_span") as mock_start_span,
+        patch("mlflow.tracking.MlflowClient") as MockClient,
+        patch("time.sleep", return_value=None),
+    ):
+        mock_client = MockClient.return_value
+        mock_parent_span = MagicMock()
+        mock_parent_span.trace_id = "trace_123"
+        mock_parent_span.span_id = "parent_456"
+        mock_start_span.return_value.__enter__.return_value = mock_parent_span
+
+        mock_child_span = MagicMock()
+        mock_child_span.span_id = "child_789"
+        mock_client.start_span.return_value = mock_child_span
+
+        mock_workspace_client.genie._api.do.side_effect = [
+            {"status": "EXECUTING_QUERY"},
+            {"status": "COMPLETED", "attachments": [{"text": {"content": "Done"}}]},
+        ]
+
+        genie.poll_for_result("123", "456")
+
+        # Verify span was closed when transitioning to COMPLETED
+        mock_client.end_span.assert_called_once()
+        end_kwargs = mock_client.end_span.call_args[1]
+        assert end_kwargs["trace_id"] == "trace_123"
+        assert end_kwargs["span_id"] == "child_789"
+        assert end_kwargs["attributes"]["final_state"] == "EXECUTING_QUERY"
```

Comment:
can we add unit tests for more coversage on these behaviors: 
1. verify no new span is created if the same status is returned consecutively (duplicate a set of statuses in the `do.side_effect` but make sure the start_span call count is still 1
2. tests for failed terminal states - cancelled, failed, query_result_expired
3. timeout case -> make sure final state attribute includes "TIMEOUT"
4. exception handling (after adding the try-except from the above comment) -> make sure polling completes successfully even if there are mlflow tracing failures